### DISCUSSION TITLE
Two classes for the SparqlBuilder - but currently Apache licensed...

### DISCUSSION THREAD

**Discussion by reckart** [eclipse-rdf4j/rdf4j#3340] [Ideas]
Hi there,

I am using RDF4J in an Apache-licensed project [INCEpTION](https://github.com/inception-project). There are two classes in INCEpTION that I think would make nice additions to RDF4J itself:

* https://github.com/inception-project/inception/blob/main/inception/inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/querybuilder/RdfCollection.java
* https://github.com/inception-project/inception/blob/main/inception/inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/querybuilder/ValuesPattern.java

I am hesitating a bit about simply opening a PR with these classes and replacing the Apache license header with the RDF4J Eclipse license header. 

Do you think these classes would be good for RDF4J? If yes, how would you suggest to have them moved over?

Best,

-- Richard

**Comment by abrokenjester** [comment]
Those definitely look useful, yes! We'd be happy to receive these as a contribution, they look like they'd fit straight in with the existing SparqlBuilder functionality. A bit of javadoc and (dare I say it) a few unit tests, perhaps a minor addition to the documentation, and we're in business :)

(Disclaimer: I am not a legal expert)

As for licensing: if you are the author/copyright holder of the code, then there is no problem if you submit them as a new contribution to RDF4J, using the Eclipse RDF4J license headers. Contributing code to RDF4J comes down to granting a perpetual, non-exclusive license for the Eclipse Foundation to use and redistribute your code under the RDF4J project's own license terms. See https://www.eclipse.org/projects/handbook/#contributing-contributors for some more in-depth explanation of this. But suffice to say it's fine to do this, and it doesn't mean you will need to remove the code from the original project (though I guess you may want to do that anyway :)).

If you _don't_ own the copyright on those classes yourself, then you will need to seek permission from the copyright holders before submitting. Since this is a small contribution (fewer than 1000 lines of code), we are usually happy to take it on good faith that you've done this. 

This is covered by the Eclipse Contributor Agreement (ECA), which every contributor needs to sign (you've probably already done this since we've had some contributions from you in the past) and which, among other things, is a promise by you that you have the right to grant us a license to your contribution.
